citesphere/src/main/java/edu/asu/diging/citesphere/core/service/impl/CitationManager.java

```diff
@@ -379,4 +380,16 @@ public CitationPage getPrevAndNextCitation(IUser user, String groupId, String co
         }
         return result;
     }
-}
+
+    @Override
+    public List<ItemDeletionResponse> deleteCitations(IUser user, String groupId, List<String> citationIdList) throws ZoteroConnectionException, ZoteroHttpStatusException {
+        List<ItemDeletionResponse> response = zoteroManager.deleteMultipleItems(user, groupId, citationIdList, zoteroManager.getLatestGroupVersion(user, groupId));
+        for(String citationId : citationIdList) {
+            Optional<ICitation> citation = citationStore.findById(citationId);
+            if(citation.isPresent()) {
+                citationStore.delete(citation.get());
```

Comment:
you wanna make sure the deletion was successful before you delete the items locally.